src/test/java/ru/hh/career/solution/ExampleServerAwareBeanTest.java

```diff
@@ -1,39 +1,39 @@
 package ru.hh.career.solution;
 
-import java.util.function.Function;
-import javax.inject.Inject;
-import org.junit.Test;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
-import org.springframework.test.context.ContextConfiguration;
-import ru.hh.nab.starter.NabApplication;
-import ru.hh.nab.testbase.NabTestBase;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 
 import javax.ws.rs.core.Response;
 
-import static org.junit.Assert.assertEquals;
+import org.junit.jupiter.api.Test;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+import ru.hh.nab.starter.NabApplication;
+import ru.hh.nab.testbase.ResourceHelper;
+import ru.hh.nab.testbase.extensions.NabJunitWebConfig;
+import ru.hh.nab.testbase.extensions.NabTestServer;
+import ru.hh.nab.testbase.extensions.OverrideNabApplication;
 
-@ContextConfiguration(classes = TestConfig.class)
-public class ExampleServerAwareBeanTest extends NabTestBase {
+@NabJunitWebConfig(TestConfig.class)
+public class ExampleServerAwareBeanTest {
 
-  @Inject
-  private Function<String, String> serverPortAwareBean;
+  @NabTestServer(overrideApplication = SpringCtxForJersey.class)
+  ResourceHelper resourceHelper;
 
   @Test
   public void testBeanWithNabTestContext() {
-    try (Response response = createRequestFromAbsoluteUrl(serverPortAwareBean.apply("/hello")).get()) {
+    try (Response response = resourceHelper.createRequestFromAbsoluteUrl(resourceHelper.baseUrl() + "/hello").get()) {
```

Comment:
[Example](https://github.com/hhru/nuts-and-bolts/commit/ed1e508530dea8a8c410a926eed1da7331e5f389#diff-1103dc0e76be397a9219132dac14a7f6ffea56b46fd050ac0e7e4a1730e8292fL24)